### PR TITLE
refactor: clean up exports

### PR DESCRIPTION
**Pull request by m-span**
More cleanups, import organization, removal of symbol exports (pub use) where not necessary.

### CODE REVIEW

File: prqlc/prqlc-parser/src/lexer/mod.rs [hunk 1]
```diff
@@ -10,12 +6,15 @@ use chumsky::{
 use serde::{Deserialize, Serialize};
 
 use crate::error::{Error, ErrorSource, Reason, WithErrorInfo};
-use crate::lexer;
 use crate::lexer::lr::{Literal, Token, TokenKind, ValueAndUnit};
 use crate::span::Span;
 
+pub mod lr;
+#[cfg(test)]
+mod test;
```
Comment (max-sixty) [inline prqlc/prqlc-parser/src/lexer/mod.rs:14]: I think convention is to always put `mod` at the top, no?

File: prqlc/prqlc/src/sql/gen_query.rs [hunk 2]
```diff
@@ -14,11 +14,11 @@ use super::gen_projection::*;
 use super::operators::translate_operator;
 use super::srq::ast::{Cte, CteKind, RelationExpr, RelationExprKind, SqlRelation, SqlTransform};
 use super::{Context, Dialect};
-use crate::ast::generic::InterpolateItem;
 use crate::ir::pl::{JoinSide, Literal};
 use crate::ir::rq::{CId, Expr, ExprKind, RelationLiteral, RelationalQuery};
 use crate::utils::{BreakUp, Pluck};
 use crate::{Error, Result, WithErrorInfo};
+use prqlc_parser::generic::InterpolateItem;
```
Comment (max-sixty) [inline prqlc/prqlc/src/sql/gen_query.rs:21]: Yeah this sort of switch makes a lot of sense; too much indirection makes it confusing...

**Review by max-sixty — APPROVED** [review]
Great, thanks!

One comment re `mod` vs `use` ordering...

**Comment by max-sixty** [comment]
FYI not sure what's going on with codecov, please ignore the error...

File: prqlc/prqlc-parser/src/lexer/mod.rs (hunk 1 above)
Comment (m-span) [reply prqlc/prqlc-parser/src/lexer/mod.rs:14]: ...apparently not. Both RustRover's Optimize Imports and cargo+nightly fmt give this ordering (I just checked)

Comment (max-sixty) [reply prqlc/prqlc-parser/src/lexer/mod.rs:14]: https://doc.rust-lang.org/beta/style-guide/items.html

> use statements, and module declarations (mod foo;, not mod { ... }) must come before other items. Put imports before module declarations.

You're right! Weird, I think most projects I've worked on do the opposite. TIL. Let's do it

**Comment by aljazerzen** [comment]
This is great.

**Pull request merged**